Mathlib/AlgebraicGeometry/Morphisms/QuasiCompact.lean

```diff
@@ -151,113 +126,58 @@ theorem isCompact_basicOpen (X : Scheme) {U : Opens X} (hU : IsCompact (U : Set
-          QuasiCompact (pullback.snd : (𝒰.pullbackCover f).obj i ⟶ 𝒰.obj i),
-        ∀ U : Opens Y, QuasiCompact (f ∣_ U),
-        ∀ {U : Scheme} (g : U ⟶ Y) [IsOpenImmersion g],
-          QuasiCompact (pullback.snd : pullback f g ⟶ _),
-        ∃ (ι : Type u) (U : ι → Opens Y) (_ : iSup U = ⊤), ∀ i, QuasiCompact (f ∣_ U i)] :=
-  quasiCompact_eq_affineProperty.symm ▸
-    QuasiCompact.affineProperty_isLocal.targetAffineLocally_isLocal.openCover_TFAE f
-#align algebraic_geometry.quasi_compact.open_cover_tfae AlgebraicGeometry.QuasiCompact.openCover_tfae
+@[reducible]
+instance : HasAffineProperty @QuasiCompact where
+  affineProperty X _ _ _ := CompactSpace X
+  eq_targetAffineLocally' := by
+    ext X Y f
+    simp only [quasiCompact_iff_forall_affine, isCompact_iff_compactSpace, targetAffineLocally,
+      Subtype.forall]
+    rfl
+  isLocal_affineProperty := by
+    constructor
+    · apply AffineTargetMorphismProperty.respectsIso_mk <;> rintro X Y Z e _ _ H
+      exacts [@Homeomorph.compactSpace _ _ _ _ H (TopCat.homeoOfIso (asIso e.inv.1.base)), H]
+    · introv _ H
+      change CompactSpace ((Opens.map f.val.base).obj (Y.basicOpen r))
+      rw [Scheme.preimage_basicOpen f r]
+      erw [← isCompact_iff_compactSpace]
+      rw [← isCompact_univ_iff] at H
+      apply isCompact_basicOpen
+      exact H
+    · rintro X Y H f S hS hS'
+      rw [← IsAffineOpen.basicOpen_union_eq_self_iff] at hS
+      · rw [← isCompact_univ_iff]
+        change IsCompact ((Opens.map f.val.base).obj ⊤).1
+        rw [← hS]
+        dsimp [Opens.map]
+        simp only [Opens.iSup_mk, Opens.coe_mk, Set.preimage_iUnion]
+        exact isCompact_iUnion fun i => isCompact_iff_compactSpace.mpr (hS' i)
+      · exact isAffineOpen_top _
 
 theorem quasiCompact_over_affine_iff {X Y : Scheme} (f : X ⟶ Y) [IsAffine Y] :
-    QuasiCompact f ↔ CompactSpace X :=
-  quasiCompact_eq_affineProperty.symm ▸ QuasiCompact.affineProperty_isLocal.affine_target_iff f
+    QuasiCompact f ↔ CompactSpace X := by
+  rw [HasAffineProperty.iff_of_isAffine (P := @QuasiCompact)]
 #align algebraic_geometry.quasi_compact_over_affine_iff AlgebraicGeometry.quasiCompact_over_affine_iff
 
 theorem compactSpace_iff_quasiCompact (X : Scheme) :
-    CompactSpace X ↔ QuasiCompact (terminal.from X) :=
-  (quasiCompact_over_affine_iff _).symm
+    CompactSpace X ↔ QuasiCompact (terminal.from X) := by
+  rw [HasAffineProperty.iff_of_isAffine (P := @QuasiCompact)]
 #align algebraic_geometry.compact_space_iff_quasi_compact AlgebraicGeometry.compactSpace_iff_quasiCompact
 
-theorem QuasiCompact.affine_openCover_iff {X Y : Scheme.{u}} (𝒰 : Scheme.OpenCover.{u} Y)
-    [∀ i, IsAffine (𝒰.obj i)] (f : X ⟶ Y) :
-    QuasiCompact f ↔ ∀ i, CompactSpace (pullback f (𝒰.map i) : _) :=
-  quasiCompact_eq_affineProperty.symm ▸ QuasiCompact.affineProperty_isLocal.affine_openCover_iff f 𝒰
-#align algebraic_geometry.quasi_compact.affine_open_cover_iff AlgebraicGeometry.QuasiCompact.affine_openCover_iff
-
```

Comment:
Since all of the specialized auxiliary reduction lemmas are removed (which I think is very good), maybe we can add the new code (i.e. `rw [HasAffineProperty.iff_of_openCover (P := @QuasiCompact) 𝒰]`)  for a reduction like this in some docstring as an example for the usage of the `HasAffineProperty` API.

Reply:
What did come out of this?

Reply:
I think the newly added docstring of the file is enough?